### ISSUE TITLE
Snapshots listing is terrible slow

### ISSUE THREAD

**Issue by kvaps** [LINBIT/linstor-server#116]
Hi we're having couple of nodes (few hundreds if exactly), and every snapshot listing is taking ages, example:

```bash 
# time linstor s l
+------------------------------------------------------------------------------+
| ResourceName   | SnapshotName       | NodeNames      | Volumes  | State      |
|==============================================================================|
| one-image-2054 | one-vm-8600-disk-2 | m10c23, m10c24 | 0: 2 GiB | Successful |
| one-image-2055 | one-vm-8465-disk-2 | m10c21, m10c22 | 0: 2 GiB | Successful |
| one-image-2055 | one-vm-8708-disk-0 | m10c21, m10c22 | 0: 2 GiB | Successful |
| one-image-2056 | one-vm-5613-disk-2 | m10c25, m10c26 | 0: 2 GiB | Successful |
+------------------------------------------------------------------------------+

real	0m33.996s
user	0m1.457s
sys	0m0.133s
```

I'm not sure what is exactly happening on the server side, and I don't know exact API call to inspect that (see https://github.com/LINBIT/linstor-client/issues/34), but I think it performs some difficult logic on client site, eg. *for each resource definition perform the new query for snapshots*, if fact it is working really slow.

**Comment by ghernadi** [comment]
I guess this is a scaling issue which we cannot simply reproduce in our labs... could you send me a trace log from the controller during such a snapshot list call? (you should have my email by now :) )

**Comment by rp-** [comment]
I guess this isn't really a controller fault, we don't have a snapshot view api yet.
So every resource definition has to be queried for snapshots and that might be slow.

**Comment by kvaps** [comment]
That sounds reasonable, thanks for explanation!

**Comment by rp-** [comment]
should be fixed with release 1.7.0 and 1.1.0
